sup3r/pipeline/forward_pass.py

```diff
@@ -1050,26 +1026,29 @@ def run(self):
                f'{self.strategy.temporal_overlap}.')
         logger.info(msg)
         max_workers = self.pass_workers
-        data = np.zeros((self.strategy.s_enhance * self.data_shape[0],
-                         self.strategy.s_enhance * self.data_shape[1],
-                         self.strategy.t_enhance * self.data_shape[2],
-                         len(self.output_features)),
-                        dtype=np.float32)
-
+        self.data = np.zeros((self.strategy.s_enhance * self.data_shape[0],
+                              self.strategy.s_enhance * self.data_shape[1],
+                              self.strategy.t_enhance * self.data_shape[2],
+                              len(self.output_features)),
+                             dtype=np.float32)
         if max_workers == 1:
-            data = self._run_serial(data)
+            self._run_serial()
         else:
-            data = self._run_parallel(data, max_workers)
+            self._run_parallel(max_workers)
 
         logger.info('All forward passes are complete.')
-        data = data[:, :, self.ti_hr_crop_slice, :]
+        self.data = self.data[:, :, self.ti_hr_crop_slice, :]
+        t_chunk = self.strategy.t_enhance * self.strategy.fwp_chunk_shape[-1]
```

Comment:
I think we had a misunderstanding with the output write chunk shape. Chunk size should not be calculated here and passed in. It should only be a function of the dataset name and dtype. See my suggestion in the file handler module.